internal/statestore/redis_indices.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statestore
+
+import (
+	"net/url"
+
+	structpb "github.com/golang/protobuf/ptypes/struct"
+	"github.com/sirupsen/logrus"
+	"open-match.dev/open-match/internal/config"
+	"open-match.dev/open-match/pkg/pb"
+)
+
+type indexedFields struct {
+	values map[string]float64
+}
+
+func extractIndexedFields(cfg config.View, t *pb.Ticket) indexedFields {
+	result := indexedFields{
+		values: make(map[string]float64),
+	}
+
+	var indices []string
+	if cfg.IsSet("ticketIndices") {
+		indices = cfg.GetStringSlice("ticketIndices")
+	}
+
+	for _, attribute := range indices {
+		v, ok := t.GetProperties().GetFields()[attribute]
+
+		if !ok {
+			redisLogger.WithFields(logrus.Fields{
+				"attribute": attribute}).Trace("Couldn't find index in Ticket Properties")
+			continue
+		}
+
+		switch v.Kind.(type) {
+		case *structpb.Value_NumberValue:
+			result.values[rangeIndexName(attribute)] = v.GetNumberValue()
+		default:
+			redisLogger.WithFields(logrus.Fields{
+				"attribute": attribute,
+			}).Warning("Attribute indexed but is not a number.")
+		}
+	}
+
+	return result
+}
+
+type indexFilter struct {
+	name     string
+	min, max float64
+}
+
+func extractIndexFilters(p *pb.Pool) []indexFilter {
+	filters := make([]indexFilter, 0)
+
+	for _, f := range p.Filters {
+		filters = append(filters, indexFilter{
+			name: rangeIndexName(f.Attribute),
+			min:  f.Min,
+			max:  f.Max,
+		})
+	}
+
+	return filters
+}
+
+func extractDeindexFilters(cfg config.View) []string {
+	var indices []string
+	if cfg.IsSet("ticketIndices") {
+		indices = cfg.GetStringSlice("ticketIndices")
+	}
+
+	var result []string
+
+	for _, index := range indices {
+		result = append(result, rangeIndexName(index))
+	}
+
+	return result
+}
+
+func rangeIndexName(attribute string) string {
```

Comment:
Can you please add a comment on why is "ri$" being appended to the attribute name? I am unsure of how this works? a comment somewhere on rangeindexname would be helpful?

Reply:
Added a comment in https://github.com/googleforgames/open-match/pull/785